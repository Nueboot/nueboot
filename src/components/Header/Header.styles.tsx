import styled from "styled-components";
import { Link } from "react-router-dom";

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-content: center;
  padding-bottom: 20px;
  /* border-bottom: 1px solid #e4e4e4; */
`;

export const Links = styled.ul`
  display: inline;
  margin: 0;
  padding: 0;

  li {
    color: #888;
    display: inline;
    list-style-type: none;
    margin-right: 30px;

    &:last-child {
      margin-right: 0px;
    }

    a {
      color: #888;
    }
  }

  a:hover {
    text-decoration: underline;
    color: black;
  }
`;

export const LogoContainer = styled(Link)`
  img {
    height: 16px;
  }
`;
