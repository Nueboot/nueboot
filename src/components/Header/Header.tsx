import React from 'react';
import { ModalTypes } from 'state/Modal/types';

import { Container, Logo, Text } from '../Styled';
import { NavigationLink } from '../Styled';

export interface DispatchProps {
  readonly openModal: (type: ModalTypes) => void;
  logout(): void;
}

export interface StateProps {
  readonly loggedIn: boolean;
  readonly pending: boolean;
}

export type HeaderProps = DispatchProps & StateProps;

const Header: React.SFC<HeaderProps> = props => {
  const handleClick = (type: ModalTypes) => () => {
    props.openModal(type);
  };

  const logout = () => {
    props.logout();
  };

  const auth = () => {
    const classList = 'pointer fw5';
    if (props.pending) {
      return null;
    }
    if (props.loggedIn) {
      return (
        <Text className={classList} onClick={logout}>
          Log Out
        </Text>
      );
    }
    return (
      <Text className={classList} onClick={handleClick('login')}>
        Log In
      </Text>
    );
  };

  return (
    <div className="header">
      <Container className="mv4 flex justify-between items-center">
        <div className="flex items-center">
          <NavigationLink to="/">
            <Logo />
          </NavigationLink>
          <NavigationLink to="/boots" className="link no-underline ml4">
            <Text className="fw5 pointer">Boots</Text>
          </NavigationLink>
        </div>
        {auth()}
      </Container>
    </div>
  );
};

export default Header;
