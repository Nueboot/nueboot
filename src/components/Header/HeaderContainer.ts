import { connect } from 'react-redux';

import { openModal } from 'state/Modal/actions';
import { ModalTypes } from 'state/Modal/types';
import { logoutUser } from 'state/Session/actions';
import { isLoggedIn, isPending } from 'state/Session/selectors';
import { State } from 'state/types';

import Header, { DispatchProps, StateProps } from './Header';

export const mapStateToProps = (state: State): StateProps => ({
  loggedIn: isLoggedIn(state),
  pending: isPending(state),
});

export const mapDispatchToProps = (dispatch): DispatchProps => ({
  openModal: (modalType: ModalTypes) => dispatch(openModal(modalType)),
  logout: () => dispatch(logoutUser()),
});

const HeaderContainer = connect(mapStateToProps, mapDispatchToProps)(Header);

export default HeaderContainer;
