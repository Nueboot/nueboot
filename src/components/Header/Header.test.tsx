import { RouteComponentProps } from '@reach/router';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Header from './Header';

describe('Header', () => {
  let component: ShallowWrapper<RouteComponentProps>;

  beforeEach(() => {
    component = shallow(<Header />);
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
