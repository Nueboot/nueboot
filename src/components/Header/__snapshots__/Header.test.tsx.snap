// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<styled.div
  className="header"
>
  <styled.div
    className="left"
  >
    <ForwardRef
      to="/"
    >
      <styled.div
        className="logo"
      >
        <Logo />
      </styled.div>
    </ForwardRef>
    <ForwardRef
      to="boots"
    >
      Boots
    </ForwardRef>
  </styled.div>
  <styled.div
    className="right"
  >
    <Auth />
  </styled.div>
</styled.div>
`;
